packages/mobile/src/exchange/ExchangeReview.tsx

```diff
@@ -1,133 +1,224 @@
 import { connect } from 'react-redux'
 import CeloAnalytics from 'src/analytics/CeloAnalytics'
 import { CustomEventNames } from 'src/analytics/constants'
 import componentWithAnalytics from 'src/analytics/wrapper'
+import FeeIcon from 'src/components/FeeIcon'
 import { exchangeTokens, fetchExchangeRate } from 'src/exchange/actions'
-import ExchangeConfirmationCard from 'src/exchange/ExchangeConfirmationCard'
 import { ExchangeRatePair } from 'src/exchange/reducer'
-import { CURRENCY_ENUM as Token } from 'src/geth/consts'
+import { CURRENCY_ENUM } from 'src/geth/consts'
 import { Namespaces } from 'src/i18n'
-import { navigate, navigateBack } from 'src/navigator/NavigationService'
+import { exchangeHeader } from 'src/navigator/Headers'
+import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { RootState } from 'src/redux/reducers'
 import { isAppConnected } from 'src/redux/selectors'
 import DisconnectBanner from 'src/shared/DisconnectBanner'
-import {
-  getNewDollarBalance,
-  getNewGoldBalance,
-  getRateForMakerToken,
-  getTakerAmount,
-} from 'src/utils/currencyExchange'
+import { getRateForMakerToken, getTakerAmount } from 'src/utils/currencyExchange'
 import { getMoneyDisplayValue } from 'src/utils/formatting'
 
 interface StateProps {
-  dollarBalance: string | null
-  goldBalance: string | null
   exchangeRatePair: ExchangeRatePair | null
   fee: string
   appConnected: boolean
 }
 
-interface ConfirmationInput {
-  makerAmount: BigNumber
-  makerToken: Token
-}
-
 interface DispatchProps {
   fetchExchangeRate: typeof fetchExchangeRate
   exchangeTokens: typeof exchangeTokens
 }
 
+interface NavProps {
+  exchangeInput: {
+    makerToken: CURRENCY_ENUM
+    makerTokenBalance: string
+    inputToken: CURRENCY_ENUM
+    inputTokenCode: string
+    inputAmount: BigNumber
+  }
+}
+
+interface State {
+  makerToken: CURRENCY_ENUM
+  inputToken: CURRENCY_ENUM
+  inputTokenCode: string
+  inputAmount: BigNumber
+}
+
 type Props = StateProps & WithNamespaces & DispatchProps & NavigationInjectedProps
 
 const mapStateToProps = (state: RootState): StateProps => ({
-  goldBalance: state.goldToken.balance,
-  dollarBalance: state.stableToken.balance,
   exchangeRatePair: state.exchange.exchangeRatePair,
   fee: getMoneyDisplayValue(0),
   appConnected: isAppConnected(state),
 })
 
-class ExchangeReview extends React.Component<Props> {
-  static navigationOptions = { header: null }
+export class ExchangeReview extends React.Component<Props, State> {
+  static navigationOptions = ({ navigation }: NavigationInjectedProps<NavProps>) => {
+    const { makerToken, makerTokenBalance } = navigation.getParam('exchangeInput')
+    return {
+      ...exchangeHeader(makerToken, makerTokenBalance),
+    }
+  }
+
+  state: State = {
+    makerToken: CURRENCY_ENUM.GOLD,
+    inputToken: CURRENCY_ENUM.GOLD,
+    inputTokenCode: this.props.t('global:gold'),
```

Comment:
Is the tokenCode a value used for display purposes? If so, maybe rename to something more clear. `inputTokenDisplayName`?

Reply:
Also, though it's probably fine in this case, using props in state initialization like this is generally considered an anti-pattern:
https://medium.com/@justintulk/react-anti-patterns-props-in-initial-state-28687846cc2e

Reply:
Thanks for linking this- seems like it's okay here as we don't expect the translation function to change, but good to know for elsewhere